### ISSUE TITLE
core: does not format user's data that include placeholder-like string

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#119]
Curentlly, This test does not be passed.

https://github.com/secretlint/secretlint/blob/c2445ccd7db13c559bff269b8e5d2dfda1936200/packages/%40secretlint/core/test/helper/SecretLintRuleMessageTranslator.test.ts#L80-L101

```diff
AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
+ actual - expected ... Lines skipped

  {
    data: {
...
      v3: '{{v1}} {{v2}} {{v3}}'
    },
+   message: '{{v1}} {{v1}} {{v2}} {{v1}} {{v2}} {{v3}} {{v1}} {{v2}} {{v3}} {{v1}} {{v2}} {{v1}} {{v2}} {{v3}} {{v1}} {{v2}} {{v3}} message',
-   message: '{{v1}} {{v2}} {{v3}} {{v1}} {{v2}} {{v3}} {{v1}} {{v2}} {{v3}} message',
    messageId: 'messageId'
  }
```

We want to fix this.

**Comment by munierujp** [comment]
How about using [Mustache](https://github.com/janl/mustache.js/) for replace placeholders?

```ts
const formatMessage = (message: string, data?: {}): string => {
    assertPlaceholder(message, data);
    if (typeof data !== "object" || data === null) {
        return message;
    }
    return Mustache.render(message, data);
}z
```

**Comment by azu** [comment]
Yes, Using template library is one of solutions.
But, [mustache.js](https://github.com/janl/mustache.js) and [handlebars.js](https://github.com/wycats/handlebars.js) has too much functions and these are for HTML.

```js
const Mustache = require("mustache");
var view = {
  title: "<div>html</div>",
  calc: function () {
    return 2 + 4;
  }
};

var output = Mustache.render("{{title}} spends {{calc}}", view);
console.log(output); // "&lt;div&gt;html&lt;&#x2F;div&gt; spends 6"
```

Also, developer friendly function is important like `assertPlaceholder`.

**Comment by azu** [comment]
📝 Maybe, We can move the run time check `assertPlaceholder` to static check by change translate function.  It is not related this issue.

```ts
// illust
const mesages = {
  hi: (props: {a: string}) => `text ${props.a}`,
  hello: {
    en: (props: {b: string}) => `text ${props.b}`,
    ja:  (props: {b: string}) => `text ${props.b}`,
  }
}

translate(mesages.hi, {
  a: "name"
});
translate(mesages.hello, {
  b: "name"
});
```

**Comment by azu** [comment]
We have introduced that static check data instead of Dynamic check data #127 
As a results, We've removed `assertPlaceholder`
